src/gui/windows/program/programmodel.cpp

```diff
@@ -129,15 +131,19 @@ QVariant ProgramModel::data(const QModelIndex &index, int role) const {
 
 void ProgramModel::setup(machine::Machine *machine) {
     this->machine = machine;
+    if (machine && machine->config().get_vm_enabled()
+        && machine->config().get_vm_mode() == machine::MachineConfig::VM_SV32) {
+        index0_offset = machine::Address(machine->config().get_va_base_addr());
```

Comment:
I think this place is a really good ilustration why separate virtual memory type will increase clarity a lot.